### PR TITLE
RFC: INT-2314 bring s3-resource-simple under CI

### PR DESCRIPTION
**Pull request by marco-m**
This works but I am concerned about the subtleties.

I can explain this face to face over the pipeline.

This provides full automatic Docker image creation and supports making changes on a feat branch pipeline without changing the production "latest" s3-resource-simple docker image.

Each time a feat branch is merged, the pipeline picks up the changes and creates a new latest image.

This PR is focused on bringing under CI the code and finding a fully automated workflow. What is missing are reasonable tests for the Concourse resource itself.

pipeline:
https://builder.ci.pix4d.com/teams/developers/pipelines/s3-resource-simple-INT-2314-resource-broken-traffic

Rendered README with the explanation of how the pipeline works:
https://github.com/Pix4D/s3-resource-simple/tree/INT-2314-resource-broken-traffic

### CODE REVIEW

File: ci/pipelines/s3-resource-simple-template.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  source:
+    repository: docker.internal:5000/((s3-resource-simple-registry))
+    insecure_registries: ['docker.internal:5000']
+    username: ((docker_cloud_pix4d_user))
+    password: ((docker_cloud_pix4d_password))
+
+- name: bucket.s3simple
+  type: s3-resource-simple
+  source:
+    <<: *s3-params
```
Comment (odormond) [inline ci/pipelines/s3-resource-simple-template.yml]: This seems to be the only place where this anchor is used so I don't think it is really helpful to add the indirection, it just make the pipeline harder to read (OK, not by much).

**Comment by kmdouglass** [comment]
Thanks @marco-m ! Do you have a link to a pipeline that we can look at?

**Review by wagdav — APPROVED** [review]
For me this can be merged as is.  This will serve as a good basis for baking docker images.  We can work out the details later.

File: ci/pipelines/s3-resource-simple-template.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    ignore_paths: [ci/work-img]
+    branch: ((branch))
+
+- name: s3-resource-simple-work-img.git
+  type: git
+  source:
+    uri: https://github.com/Pix4D/s3-resource-simple.git
+    paths: [ci/work-img]
+    branch: ((branch))
+
```
Comment (kmdouglass) [inline ci/pipelines/s3-resource-simple-template.yml]: Would you mind adding just a short comment about what the semantic differences are between s3-resource-simple-work-img.docker and s3-resource-final-img.docker? Without any context I am not sure how they are to be used.

**Review by kmdouglass — APPROVED** [review]

File: ci/pipelines/s3-resource-simple-template.yml (hunk 2 above)
Comment (odormond) [reply ci/pipelines/s3-resource-simple-template.yml]: Even better, name that entirely differently. My understanding is that this is the docker image used to run the test script that just produces a tree of files. It has nothing to do with S3. So my proposal is to name it like `test-fs-tree-producer.docker`.

**Review by odormond — CHANGES_REQUESTED** [review]

**Review by aledegano — APPROVED** [review]

**Review by wagdav — APPROVED** [review]

**Pull request merged**